astropy/io/misc/tests/test_yaml.py

```diff
@@ -316,3 +319,57 @@ def test_yaml_load_of_object_arrays_fail():
             order: C
             shape: !!python/tuple [3]"""
         )
+
```

Comment:
This regression test is fine but more importantly there should be YAML-level unit tests covering all the cases in the new code. I.e. serializing a simple dict structure with all the impacted data types and confirming the output and correct round-trip.

Reply:
Gotcha. I'll work on it.

Reply:
Actually the unit tests in question already exists, I just needed to add a pytest fixture to compensate for global settings (see #15096).
They also capture the regressions properly, so maybe we should just ditch the more complex integration test, if that's okay with you.

Reply:
Excellent about the tests already being there!  And yes, I'm good with dropping the integration test.

Reply:
great, I rebased and greatly simplified the branch !